crates/networking/discv5/src/subnet.rs

```diff
@@ -1,173 +1,452 @@
+        attestation_subnets.enable_attestation_subnet(1).unwrap();
+        attestation_subnets.enable_attestation_subnet(5).unwrap();
+
+        let mut sync_committee_subnets = SyncCommitteeSubnets::new();
+        sync_committee_subnets
+            .enable_sync_committee_subnet(0)
+            .unwrap();
+        sync_committee_subnets
+            .enable_sync_committee_subnet(2)
+            .unwrap();
+
+        let enr = Enr::builder()
+            .add_value(ATTESTATION_BITFIELD_ENR_KEY, &attestation_subnets)
+            .add_value(SYNC_COMMITTEE_BITFIELD_ENR_KEY, &sync_committee_subnets)
+            .build(&combined_key)
+            .expect("Failed to build ENR");
+
+        let decoded_attnets = enr
+            .get_decodable::<AttestationSubnets>(ATTESTATION_BITFIELD_ENR_KEY)
+            .expect("Failed to get attestation bitfield")
+            .expect("Failed to decode attestation bitfield");
+
+        assert!(decoded_attnets.get(1).unwrap());
+        assert!(decoded_attnets.get(5).unwrap());
+        assert!(!decoded_attnets.get(0).unwrap());
+        assert!(!decoded_attnets.get(10).unwrap());
+
+        let decoded_syncnets = enr
+            .get_decodable::<SyncCommitteeSubnets>(SYNC_COMMITTEE_BITFIELD_ENR_KEY)
+            .expect("Failed to get sync committee bitfield")
+            .expect("Failed to decode sync committee bitfield");
+
+        assert!(decoded_syncnets.get(0).unwrap());
+        assert!(decoded_syncnets.get(2).unwrap());
+        assert!(!decoded_syncnets.get(1).unwrap());
+        assert!(!decoded_syncnets.get(3).unwrap());
+    }
+
+    #[test]
+    fn test_new_subnet_types() {
+        let secret_key = SigningKey::random(&mut rand::thread_rng());
+        let combined_key = CombinedKey::from(secret_key);
+
+        let mut attestation_subnets = AttestationSubnets::new();
+        attestation_subnets.enable_attestation_subnet(3).unwrap();
+        attestation_subnets.enable_attestation_subnet(42).unwrap();
+
+        let mut sync_committee_subnets = SyncCommitteeSubnets::new();
+        sync_committee_subnets
+            .enable_sync_committee_subnet(0)
+            .unwrap();
+        sync_committee_subnets
+            .enable_sync_committee_subnet(2)
+            .unwrap();
+
+        let enr = Enr::builder()
+            .add_value(ATTESTATION_BITFIELD_ENR_KEY, &attestation_subnets)
+            .add_value(SYNC_COMMITTEE_BITFIELD_ENR_KEY, &sync_committee_subnets)
+            .build(&combined_key)
+            .expect("Failed to build ENR");
+
+        let decoded_attnets = enr
+            .get_decodable::<AttestationSubnets>(ATTESTATION_BITFIELD_ENR_KEY)
+            .expect("Failed to get attestation subnets")
+            .expect("Failed to decode attestation subnets");
+
+        assert!(decoded_attnets.get(3).unwrap());
+        assert!(decoded_attnets.get(42).unwrap());
+        assert!(!decoded_attnets.get(0).unwrap());
+        assert!(!decoded_attnets.get(10).unwrap());
```

Comment:
```suggestion
        assert!(decoded_attnets.get(3).unwrap());
        assert!(decoded_attnets.get(42).unwrap());
        assert!(!decoded_attnets.get(0).unwrap());
        assert!(!decoded_attnets.get(10).unwrap());
```

could we use assert_eq here and remove the unwrap